test/functional/core/spellfile_spec.lua

```diff
@@ -117,4 +117,26 @@ describe('spellfile', function()
     local fname = fn.fnamemodify(testdir .. '/spell/spell.add', ':p')
     api.nvim_set_option_value('spellfile', fname, {})
   end)
+
+  describe('default location', function()
+    it("is stdpath('data')/spell/en.utf-8.add", function()
+      n.command('set spell')
+      n.insert('abc')
+      n.feed('zg')
+      eq(testdir .. '/xdg_data/nvim/spell/en.utf-8.add', api.nvim_get_option_value('spellfile', {}))
+    end)
+
+    it("is not set if stdpath('data') is not writable", function()
+      n.command('set spell')
+      vim.uv.fs_mkdir(testdir .. '/xdg_data', 444)
```

Comment:
I think chmod 444 isn't possible on Windows. Instead, to make this test work on windows, try overwriting `.../xdg_data` with a file instead of a directory. That should hopefully "fail" correctly.

If that doesn't work, then use `t.is_os('win')` to `return` from this test (don't use `skip()` because it's not possible to ever "fix" this).